packages/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts

```diff
@@ -234,6 +234,42 @@ export const useGridCellEditing = (
     [apiRef],
   );
 
+  const runIfNoPreProcessError =
+    <Args extends any[]>(callback?: (...args: Args) => void) =>
+    async (...args: Args) => {
+      if (callback) {
+        const { field, id, value, debounceMs, unstable_skipValueParser: skipValueParser } = args[0];
+        const column = apiRef.current.getColumn(field);
+        const row = apiRef.current.getRow(id)!;
+
+        let parsedValue = value;
+        if (column.valueParser && !skipValueParser) {
+          parsedValue = column.valueParser(value, row, column, apiRef);
+        }
+
+        const editingState = gridEditRowsStateSelector(apiRef.current.state);
+        let newProps: GridEditCellProps = {
+          ...editingState[id][field],
+          value: parsedValue,
+          changeReason: debounceMs ? 'debouncedSetEditCellValue' : 'setEditCellValue',
+        };
+
+        if (column.preProcessEditCellProps) {
+          const hasChanged = value !== editingState[id][field].value;
+
+          newProps = { ...newProps, isProcessingProps: true };
```

Comment:
This entire logic was copied from https://github.com/mui/mui-x/blob/aed8aeb903ef7d319419a7c50eb44286d547888d/packages/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts#L449-L469

Thought of storing `{ id, row, props: newProps, hasChanged }` in ref, which are computed in below function, but wasn't sure if `setCellEditingEditCellValue` is run always and runs before `runIfNoPreProcessError` https://github.com/mui/mui-x/blob/aed8aeb903ef7d319419a7c50eb44286d547888d/packages/x-data-grid/src/hooks/features/editing/useGridCellEditing.ts#L445

Reply:
Thanks!
Did you consider following the same approach as in https://github.com/mui/mui-x/pull/11383

Reply:
Wasn't aware of that PR. Now I've updated code to use same approach as in https://github.com/mui/mui-x/pull/11383.